dialogy/base/output/__init__.py

```diff
@@ -44,110 +44,105 @@
 """
 from __future__ import annotations
 
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Dict, List, Union, Optional
 
-import attr
+from pydantic import BaseModel, Field, validator
 
 from dialogy import constants as const
 from dialogy.types import BaseEntity, Intent
+from dialogy.types.entity.deserialize import EntityDeserializer
+
 
 JSON_LIST_TYPE = List[Dict[str, Any]]
 ORIGINAL_INTENT_TYPE = Dict[str, Union[str, float]]
 
 
-@attr.frozen
-class Output:
+class Output(BaseModel):
     """
     Represents output of the SLU API.
     """
 
-    intents: List[Intent] = attr.ib(default=attr.Factory(list), kw_only=True)
+    intents: List[Intent] = Field(default_factory=list)
     """
     A list of intents. Produced by :ref:`XLMRMultiClass <XLMRMultiClass>`
     or :ref:`MLPMultiClass <MLPMultiClass>`.
     """
 
-    entities: List[BaseEntity] = attr.ib(default=attr.Factory(list), kw_only=True)
+    entities: List[BaseEntity] = Field(default_factory=list)
     """
     A list of entities. Produced by :ref:`DucklingPlugin <DucklingPlugin>`, 
     :ref:`ListEntityPlugin <ListEntityPlugin>` or :ref:`ListSearchPlugin <ListSearchPlugin>`.
     """
 
-    original_intent: ORIGINAL_INTENT_TYPE = attr.ib(
-        default=attr.Factory(dict), kw_only=True
-    )
+    original_intent: ORIGINAL_INTENT_TYPE = Field(default_factory=dict)
 
-    @intents.validator  # type: ignore
-    def _are_intents_valid(
-        self, _: attr.Attribute, intents: List[Intent]  # type: ignore
-    ) -> None:
-        if not isinstance(intents, list):
-            raise TypeError(f"intents must be a list, not {type(intents)}")
+    duration: float = Field(default=0.0)
 
-        if not intents:
-            return
+    def __init__(self, **data: Any):
+        if "entities" in data:
+            entities = [
+                EntityDeserializer.deserialize_json(**ent) for ent in data["entities"]
+            ]
```

Comment:
This does not handle the case where `data[entities]` is a list of entity objects directly... `EntityDeserializer.deserialize_json(**ent)` will fail there (some tests fail because of that)